test/connectors/test_torch_connector.py

```diff
@@ -659,6 +659,205 @@ def test_circuit_qnn_batch_gradients(self, config):
             places=4,
         )
 
+    @requires_extra_library
+    def test_opflow_qnn_hybrid_batch_gradients(self):
+        """Test gradient back-prop for batch input in hybrid opflow qnn."""
+
+        try:
+            from torch import cat
+            from torch.nn import Linear, Module
+            import torch.nn.functional as F
+        except ImportError as ex:
+            raise MissingOptionalLibraryError(
+                libname="Pytorch",
+                name="TorchConnector",
+                pip_install="pip install 'qiskit-machine-learning[torch]'",
+            ) from ex
+
+        num_inputs = 2
+
+        # set up QNN
+        qnn = TwoLayerQNN(
+            num_qubits=num_inputs,
+            quantum_instance=self.sv_quantum_instance,
+            input_gradients=True,  # for hybrid qnn
+        )
+
+        # set up dummy hybrid PyTorch module
+        class Net(Module):
+            """Pytorch nn module."""
+
+            def __init__(self):
+                super().__init__()
+                self.fc1 = Linear(4, 2)
+                self.qnn = TorchConnector(
+                    qnn, np.array([0.1] * qnn.num_weights)
+                )  # Apply torch connector
+                self.fc2 = Linear(1, 1)  # 1-dimensional output from QNN
+
+            def forward(self, x):
+                """Forward pass."""
+                x = F.relu(self.fc1(x))
+                x = self.qnn(x)  # apply QNN
+                x = self.fc2(x)
+                return cat((x, 1 - x), -1)
+
+        model = Net()
+
+        # random data set
+        x = Tensor(np.random.rand(5, 4))
+        y = Tensor(np.random.rand(5, 2))
+
+        # define optimizer and loss
+        optimizer = SGD(model.parameters(), lr=0.1)
+        f_loss = MSELoss(reduction="sum")
+
+        # loss and gradients without batch
+        optimizer.zero_grad(set_to_none=True)
+        sum_of_individual_losses = 0.0
+        for x_i, y_i in zip(x, y):
+            output = model(x_i)
+            sum_of_individual_losses += f_loss(output, y_i)
+        sum_of_individual_losses.backward()
+        sum_of_individual_gradients = 0.0
+        for n, param in model.named_parameters():
+            name = n.split(".")
+            # make sure gradient is not None
+            self.assertFalse(param.grad is None)
+            if name[1] == "weight" or name[1] == "_weights":
```

Comment:
I think we can replace `name = n.split(".")`, `name[1] == "weight"` with `n.endwith(".weight")` once we have identical names for weights.

Reply:
Yes, once the names are fixed, this line can be simplified.